### PR TITLE
MSBuild task has been updated - build log file will be created only when createLogFile = true

### PR DESCRIPTION
**Pull request by andy-mishechkin**
**Task name**: MSBuild

**Description**: In previous version log file was created in two cases: System.Debug = true, or createLogFile = true, which caused certain issues (see the related ticket). Now log file will created only when createLogFile = true is specified in MSBuild task.

**Documentation changes required:** N

**Added unit tests:** N

**Attached related issue:** https://github.com/microsoft/build-task-team/issues/953

**Checklist**:
- [X] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [X] Checked that applied changes work as expected
Links to the test pipelines:
https://dev.azure.com/v-andreym/MSBuild%20Test%20Project/_build?definitionId=1
https://dev.azure.com/v-andreym/MSBuild%20Test%20Project/_build?definitionId=2

### CODE REVIEW

File: Tasks/MSBuildV1/package-lock.json [hunk 1]
```diff
@@ -321,9 +321,9 @@
       },
       "dependencies": {
         "azure-pipelines-task-lib": {
-          "version": "2.11.3",
-          "resolved": "https://registry.npmjs.org/azure-pipelines-task-lib/-/azure-pipelines-task-lib-2.11.3.tgz",
-          "integrity": "sha512-gXops6Npkloh7AKPIvptx0KY1kdf1yd+BFdp/ksnbyW7CS3uXzBQXU1Dermr7A895TrOd02PI6hmsL8cBlePNA==",
+          "version": "2.12.2",
```
Comment (anatolybolshakov) [inline Tasks/MSBuildV1/package-lock.json:324]: Is this expected change? There is no related changes in package.json

File: Tasks/MSBuildV1/Tests/PassesArguments.ps1 [hunk 2]
```diff
@@ -44,7 +44,7 @@ foreach ($variableSet in $variableSets) {
     Register-Mock Select-MSBuildPath { 'Some location' } -- -Method 'Some input method' -Location 'Some input location' -PreferredVersion 'Some input version' -Architecture 'Some input architecture'
     Register-Mock Invoke-BuildTools { 'Some build output line 1', 'Some build output line 2' }
 
-    $ExpectedCreateLogFile = if ($variableSet.Debug) { $true } else { $variableSet.CreateLogFile }
+    $ExpectedCreateLogFile = { $variableSet.CreateLogFile }
```
Comment (anatolybolshakov) [inline Tasks/MSBuildV1/Tests/PassesArguments.ps1:47]: Nit: do we need curly braces?

File: Tasks/MSBuildV1/MSBuild.ps1 [hunk 3]
```diff
@@ -19,7 +19,7 @@ try {
     [bool]$maximumCpuCount = Get-VstsInput -Name MaximumCpuCount -AsBool
     [bool]$restoreNuGetPackages = Get-VstsInput -Name RestoreNuGetPackages -AsBool
     [bool]$logProjectEvents = Get-VstsInput -Name LogProjectEvents -AsBool
-    [bool]$createLogFile = (Get-VstsInput -Name CreateLogFile -AsBool) -or $debug
+    [bool]$createLogFile = (Get-VstsInput -Name CreateLogFile -AsBool)
```
Comment (anatolybolshakov) [inline Tasks/MSBuildV1/MSBuild.ps1:22]: Could you please apply the same changes for VSBuild task also?

File: Tasks/MSBuildV1/Tests/PassesArguments.ps1 (hunk 2 above)
Comment (andy-mishechkin) [reply Tasks/MSBuildV1/Tests/PassesArguments.ps1]: Nope
I've fixed test

File: Tasks/MSBuildV1/MSBuild.ps1 (hunk 3 above)
Comment (andy-mishechkin) [reply Tasks/MSBuildV1/MSBuild.ps1:22]: Done and tested:
https://dev.azure.com/v-andreym/MSBuild%20Test%20Project/_build/results?buildId=93&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9
https://dev.azure.com/v-andreym/MSBuild%20Test%20Project/_build/results?buildId=95&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9

File: Tasks/MSBuildV1/package-lock.json (hunk 1 above)
Comment (andy-mishechkin) [reply Tasks/MSBuildV1/package-lock.json]: package-lock.json has been reverted from master branch

**Review by anatolybolshakov — APPROVED** [review]
LGTM, thanks!

**Review by EzzhevNikita — APPROVED** [review]

**Pull request merged**